### ISSUE TITLE
Add 'csharp_style_implicit_object_creation_in_return_statements' option for 'Simplify new expression (IDE0090)' analyzer

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#85146]
**Brief description:** today, the ['Simplify new expression (IDE0090)' analyzer](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0090) only exposes the `csharp_style_implicit_object_creation_when_type_is_apparent` option. There's no way to make the analyzer also fire for return/yield statements/expressions.

We should add a new option (e.g. `csharp_style_implicit_object_creation_in_return_statements`) to support this.

This would fire the diagnostic for any `return` or `switch` statements/expressions that can be simplified as `new()`.

**Languages applicable:** C#

Is your analyzer C# only? VB Only? Or Both?

**Code example that the analyzer should report:**

```csharp
Foo M()
{
    return new Foo(); // IDE0090, can simplify to 'return new();'
}

IEnumerable<Foo> M()
{
    yield return new Foo(); // IDE0090, can simplify to 'yield return new();'
}
```

Same for when using expressions as well.

**Documentation requirements:**

The analyzer already has docs ([here](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0090)), we'd just need to document the new option.

**Comment by CyrusNajmabadi** [comment]
i don't like this.  instead ot this approach, i would extend csharp_style_implicit_object_creation_when_type_is_apparent to be more than true/false, and allow for specifying well-known contexts that are considered to be 'apparent'. so `return_statement_expression` and `yield_statement_expression` could be well known contexts.  that sort of thing.

**Comment by Sergio0694** [comment]
That seems fine. I don't have a strong preference for the exact shape, I just think the general feature is worth having 🙂

**Comment by CyrusNajmabadi** [comment]
I came here for an argument!
